packages/sdk/src/evm/core/classes/erc-721.ts

```diff
@@ -198,18 +199,14 @@ export class Erc721<
    * ```
    * @twfeature ERC721
    */
-  public async transfer(
-    to: string,
-    tokenId: BigNumberish,
-  ): Promise<TransactionResult> {
+  transfer = transaction(async (to: string, tokenId: BigNumberish) => {
     const from = await this.contractWrapper.getSignerAddress();
-    return {
-      receipt: await this.contractWrapper.sendTransaction(
-        "safeTransferFrom(address,address,uint256)",
-        [from, to, tokenId],
-      ),
-    };
-  }
+    return Transaction.fromContractWrapper({
```

Comment:
curious - did you check the `TransactionTask` class? was that not reusable?

Reply:
yep, wanted to make this separate as it's designed differently + easier for now since I don't have to replace all the `TransactionTask` for testing. Also prefer `Transaction` name, curious on your preference.